Review and Update Developing Poly doc

lread opened the pull request:
Re-order and re-section some content for an easier to follow flow.

Add a Contributing section.
Currently my (@lread) preferences and biases, polylith core team can adapt to their point of view.

Add a few more details on cljdoc.

Give concrete example of using other workspace usage.

Edits for clarity.

lread committed:
Review and Update Developing Poly doc

Re-order and re-section some content for an easier to follow flow.

Add a Contributing section.
Currently my (@lread) preferences and biases, polylith core team can
adapt to their point of view.

Add a few more details on cljdoc.

Give concrete example of using other workspace usage.

Edits for clarity.

tengstrand commented on doc/developing-poly.adoc:
```diff
@@ -1,102 +1,136 @@
+=== `fake-poly`
 
-This argument is used to fake that we are using a non-snapshot version of the `poly` tool, and not the `polyx` tool.
+Fake that you are using a stable production release of `poly` when you might actually be using a `SNAPSHOT` release, ' polyx`, or both.
```
Change ' to ` for polyx

lread replied:
tx!

lread replied:
fixed

tengstrand commented on doc/developing-poly.adoc:
```diff
@@ -1,102 +1,136 @@
-poly help :fake-poly
-# Excludes "-SNAPSHOT" from the Polylith ascii art logo:
-poly :fake-poly
+poly help :fake-polya # <1>
```
Should be :fake-poly I guess.

lread replied:
tx!

lread replied:
vi keys get me sometimes, fixed!

tengstrand commented on doc/developing-poly.adoc:
```diff
@@ -145,58 +237,66 @@ Among other https://asciidoctor.org/docs/asciidoc-recommended-practices/#one-sen
 ** `top-ns:my-ns`, `branch:my-branch` and `:commit` as arguments
-* Always link to referenced commands and concepts to make it easier for readers to explore and dig into details.
+* Link to referenced commands and concepts to make it easier for readers to explore and dig into details.
 * Use "we" when talking from the Polylith core team and "you" when directing the reader through some steps or a tutorial.
```
Here somewhere in this list we could tell people to use special syntax for keyboard shortcuts, found [here](https://cljdoc.org/d/polylith/clj-poly/0.2.18-SNAPSHOT/doc/shell#quit).

lread replied:
will do

lread replied:
done!

tengstrand commented on doc/developing-poly.adoc:
```diff
@@ -1,102 +1,136 @@
+
+Please never force push on your PR, as this makes reviewing incremental changes impossible for us.
+When we merge your PR, we'll usually squash it, so that will clean up any rambling work-in-progress.
+
```
Tell people to add a row in [next-release.txt](https://github.com/polyfy/polylith/blob/master/next-release.txt) when they create a PR. Additional info could be that this file is cleared when we make a final/stable release.

lread replied:
yep, can do

lread replied:
Added note

tengstrand reviewed (changes requested):
I really like your changes. Awesome job!

lread committed:
Address review feeback

Fix typos.

Contributing: describe next-release.txt.

Note that adoc kbd macros should be used.
